plugins/qcheck-stm/doc/index.mld

```diff
@@ -141,15 +141,59 @@
 [ensures] clauses that were not used for the [next_state] function but it also
 uses the [checks] clauses and the [raises] ones.
 
-Now you can generate the QCheck-STM file by running the following command where
-you indicate the file you want to test, the function call to build a value of
-the type indicated in the third argument. You can write the generated code into
-a file, using the [-o] option.
+Now you need a configuration file. Configuration files for Ortac/QCheck-STM are
+OCaml modules contaning {e at least} a type declaration for `sut` (usually the
+type `t` of the module under test with all the type parameters instantiated)
+and a value declaration for `init_sut` (a call to a function from the module
+under test returning a value of type `sut`)
+
+{@ocaml file=example_config.ml[
+open Example
+
+type sut = char t
+
+let init_sut = make 42 'a'
+]}
+
+Then, you can generate the QCheck-STM file by running the following command where
+you indicate the file you want to test and the configuration file. You can
+write the generated code into a file, using the [-o] option.
 
 {@sh set-ORTAC_ONLY_PLUGIN=qcheck-stm[
-$ ortac qcheck-stm example.mli "make 42 'a'" "char t" -o stm_example.ml
+$ ortac qcheck-stm example.mli example_config.ml -o stm_example.ml
+]}
+
+The other information you can put in the configuration file are:
+
+{ul {li custom [QCheck] generator in a [Gen] module}
+    {li custom [STM] printers in a [Pp] module}
+    {li custom [STM.ty] extensions in a [Ty] module}}
+
+Let's say for example that you want to run the generated tests using a custom
+generator for the type [int] that only generates small positive integers. You
+can still rely on the generated command generator, you just have to override the
+[int] generator:
+
+{@ocaml file=example_config_qcheck.ml[
+open Example
+
+type sut = char t
+
+let init_sut = make 42 'a'
+
+module Gen = struct
+  include QCheck.Gen
+
+  let int = small_signed_int
+end
 ]}
 
+Note that you have to include, not just open, the [QCheck.Gen] module.
```

Comment:
Really? Note that the `custom_config` test doesn’t even `open` `QCheck.Gen`, as it will be included automatically by the wrapper, won’t it?
It might be a surprising choice, though, as the configuration module is not well-formed OCaml. It might make a nicer interface to make sure that the configuration module can be type-checked on its own.